tests/SCF/test_SCF.py

```diff
@@ -1,19 +1,88 @@
 import pytest
 import SCF
+import pickle
+from mol import mol
+
+
+Suv = pickle.load(open("suv.pkl", "rb"))
+Tuv = pickle.load(open("real_tuv.pkl", "rb"))
+eri = pickle.load(open("real_eri.pkl", "rb"))
+Vuv = pickle.load(open("real_vuv.pkl", "rb"))
 
 
 def test_calc_nuclear_repulsion_energy(mol_h2o):
-    assert True
-#    assert SCF.calc_nuclear_repulsion_energy(mol_h2o) == 8.00236706181077,\
-#        "Nuclear Repulsion Energy Test (H2O) Failed"
+    assert SCF.calc_nuclear_repulsion_energy(mol_h2o) == 8.00236706181077
+    "Nuclear Repulsion Energy Test (H2O) Failed"
 
 
 def test_calc_initial_density(mol_h2o):
-    """
-    Tests that the initial density returns a zero matrix
-    and tests dimensions
-    """
-
     Duv = SCF.calc_initial_density(mol_h2o)
+    assert Duv.shape == (mol_h2o.nao, mol_h2o.nao)
     assert Duv.sum() == 0.0
-    assert Duv.shape == (mol_h2o.nao,mol_h2o.nao)
+    "Calculate Initial Density Test Failed"
+
+
+def test_calc_hcore_matrix(mol_h2o):
+    assert SCF.calc_hcore_matrix(Tuv, Vuv)[0, 0] == -32.57739541261037
+    assert SCF.calc_hcore_matrix(Tuv, Vuv)[3, 4] == 0.0
+    assert SCF.calc_hcore_matrix(Tuv, Vuv)[4, 3] == 0.0
+    "Calculate h_core Matrix Test Failed"
+
+
+def test_calc_fock_matrix(mol_h2o):
+    Duv = SCF.calc_initial_density(mol_h2o)
+    h_core = SCF.calc_hcore_matrix(Tuv, Vuv)
+
+    assert SCF.calc_fock_matrix(mol_h2o, h_core, eri, Duv)[0, 0] == \
+        pytest.approx(-32.57739541261037)
+
+    assert SCF.calc_fock_matrix(mol_h2o, h_core, eri, Duv)[2, 5] == \
+        pytest.approx(-1.6751501447185015)
+
+    assert SCF.calc_fock_matrix(mol_h2o, h_core, eri, Duv)[5, 2] == \
+        pytest.approx(-1.6751501447185015)
+    "Calculate Fock Matrix Test Failed"
+
+
+def test_solve_Roothan_equations(mol_h2o):
+    Duv = SCF.calc_initial_density(mol_h2o)
+    h_core = SCF.calc_hcore_matrix(Tuv, Vuv)
+    Fuv = SCF.calc_fock_matrix(mol_h2o, h_core, eri, Duv)
+    mo_energies, mo_coeffs = SCF.solve_Roothan_equations(Fuv, Suv)
+
+    assert mo_energies == pytest.approx([-32.57830292, -8.08153571,
+                                        -7.55008599, -7.36396923,
+                                        -7.34714487, -4.00229867, -3.98111115])
+
+    assert mo_coeffs[0, 0] == pytest.approx(-1.00154358e+00)

```

Comment:
These may fail as was noted in class and on the issue board that the sign of the values is not guaranteed from run to run so you should screen against the absolute values.